tests/unit_tests/test_formatter.py

```diff
@@ -22,3 +24,17 @@ def test_formatting():
         }
     }
     assert prepare_pin(test_place, visible_fields) == expected_data
+
+
+@pytest.mark.parametrize(
+    ("arg", "expected"),
+    (
+        ([], []),
+        (["abc"], [gettext("abc")]),
+        ({}, {}),
+        ({"a": "b"}, {"a": "b"}),
+        ("abc", gettext("abc")),
+    ),
+)
+def test_safe_gettext(arg: object, expected: object) -> None:
+    assert safe_gettext(arg) == expected
```

Comment:
This test may be good for coverage, but doesn't test anything.

It is for sure obfuscated and it's impossible to know what this function does... Which is good that you brought it to atencion. 

babels `gettext` gets string or its translation. So purpose of this was to get translations (Polish in example) `['kot', 'pies']`  from original (English example) `['cat', 'dog']`.

Don't really know what should be done here, but this code is not the answer. :/

Reply:
For sure code should be more explanatory (maybe safe_gettext is not the best name)...

We could mock some internalization or flask_babel.gettext to return translation.

Reply:
maybe we don't need such strict coverage requirement for now?

I know the test isn't perfect, but I don't think it's that important.